enterprise/frontend/src/metabase-enterprise/auth/index.ts

```diff
@@ -24,50 +24,62 @@ const SSO_PROVIDER = {
   Button: SsoButton,
 };
 
-if (hasPremiumFeature("sso_saml")) {
-  PLUGIN_AUTH_PROVIDERS.SettingsSAMLForm = SettingsSAMLForm;
-}
-
-if (hasPremiumFeature("sso_jwt")) {
-  PLUGIN_AUTH_PROVIDERS.SettingsJWTForm = SettingsJWTForm;
-}
-
+// Always set AuthSettingsPage - this doesn't depend on premium features
 PLUGIN_AUTH_PROVIDERS.AuthSettingsPage = AuthSettingsPage;
 
-PLUGIN_AUTH_PROVIDERS.providers.push((providers) => {
-  if (
-    (hasPremiumFeature("sso_jwt") || hasPremiumFeature("sso_saml")) &&
-    MetabaseSettings.get("other-sso-enabled?")
-  ) {
-    providers = [SSO_PROVIDER, ...providers];
+/**
+ * Initialize auth plugin features that depend on hasPremiumFeature.
+ */
+export function initializePlugin() {
+  if (hasPremiumFeature("sso_saml")) {
+    PLUGIN_AUTH_PROVIDERS.SettingsSAMLForm = SettingsSAMLForm;
   }
-  if (
-    hasPremiumFeature("disable_password_login") &&
-    !MetabaseSettings.isPasswordLoginEnabled() &&
-    !MetabaseSettings.isLdapEnabled()
-  ) {
-    providers = providers.filter((p) => p.name !== "password");
+
+  if (hasPremiumFeature("sso_jwt")) {
+    PLUGIN_AUTH_PROVIDERS.SettingsJWTForm = SettingsJWTForm;
   }
-  return providers;
-});
 
-if (hasPremiumFeature("disable_password_login")) {
-  PLUGIN_IS_PASSWORD_USER.push((user) =>
-    Boolean(
-      user.sso_source !== "google" &&
-        user.sso_source !== "ldap" &&
-        MetabaseSettings.isPasswordLoginEnabled(),
-    ),
-  );
-}
+  // Add provider function that handles SSO and password login
+  const enhancedProviderFunction = (providers: any) => {
```

Comment:
This was properly inferred as `AuthProvider[]`.

<img width="1426" height="232" alt="Image" src="https://github.com/user-attachments/assets/bdb90722-2b7b-408b-be64-d8da5f8b8361" />

I think we can either revert it back to belong inside `.push()` for auto type inference, or we properly annotate this explicit:

```suggestion
  const enhancedProviderFunction = (providers: AuthProvider[]) => {
```